ash/src/extensions/khr/video_encode_queue.rs

```diff
@@ -0,0 +1,57 @@
+//! <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_video_encode_queue.html>
+
+use crate::prelude::*;
+use crate::vk;
+
+impl crate::khr::video_encode_queue::Device {
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEncodeVideoKHR.html>
+    #[doc(alias = "vkCmdEncodeVideoKHR")]
+    #[inline]
+    pub unsafe fn cmd_encode_video(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        encode_info: &vk::VideoEncodeInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_encode_video_khr)(command_buffer, encode_info)
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetEncodedVideoSessionParametersKHR.html>
+    #[doc(alias = "vkGetEncodedVideoSessionParametersKHR")]
+    #[inline]
+    pub unsafe fn get_encoded_video_session_parameters(
+        &self,
+        session_parameters_info: &vk::VideoEncodeSessionParametersGetInfoKHR<'_>,
+        info: &mut vk::VideoEncodeSessionParametersFeedbackInfoKHR<'_>,
```

Comment:
I pushed a version with a separate `_len` method for determining the length of the output slice, and optional `feedback_info` on both methods. It required a new `RawMutPtr` helper. (Happy to adjust if you want something different.)

Reply:
We discussed this function in chat. It is used to query three "distinct" things:

1. `pFeedbackInfo`;
2. `pDataSize`;
3. `pData`.

Of those, we identified two patterns that appear to be useful:

1. Query everything. This is already provided by the current implementation of this PR, and conveniently makes use of `read_into_uninitialized_vector()`. My only "optimization request" would be to pass `pFeedbackInfo` only during the first or second call, so that the driver doesn't have to (walk the optional `pNext` chain and) fill the struct(s) repeatedly;
2. Query `pFeedbackInfo` only, and optionally `pData`. This implies that the first call should provide `pDataSize` too, so that there is no third call needed for that (i.e. instead of adding a `pFeedbackInfo=NULL` variant based on `read_into_uninitialized_vector()`).

I didn't list this, but there is a third pattern that I am not sure is useful: query `pData` exclusively. This is identical to the ail of pattern 2., where a `read_into_uninitialized_vector()`-based variant would let `ash` perform the `pDataSize` query oblivious to the user.

---

Since this either spawns a lot of variants of the same function, or hides possible optimizations from the user (insofar returning preallocated `Vec`s isn't considered suboptimal...), we think having a `_len()` variant of this function - despite not needing to initialize the data - is the  solution. If both getters carry `feedback_info: Option<&mut ...>`, the caller can decide when and where to query this.

However, this implies that the array getter takes `&mut [MaybeUninit<u8>]` to not unnecessarily force the callers to (zero-)initialize a `&mut [u8]` when it is going to be overwritten anyway.

Reply:
@Ralith what is your thought on this, and the provided implementation/resolution?

Reply:
@Friz64 I believe your input is also valuable on the API of this function. See also my additional concerns in https://github.com/ash-rs/ash/pull/965#pullrequestreview-2485790116.

Reply:
Discussed this some more in chat. I don't have a ton of context for this API, or really a very strong opinion, but it seems like the current draft is close to a sweet spot, preserving consistency with two-call patterns elsewhere, not constraining callers, and not doing redundant work. I concur with @MarijnS95 that tweaking the non-`_len` version to return an `&mut [u8]` containing the initialized (sub)set of the input slice on success would be a nice convenient to add.